bundles/core/org.eclipse.smarthome.core.thing/src/main/java/org/eclipse/smarthome/core/thing/internal/CommunicationManager.java

```diff
@@ -326,6 +326,10 @@ private void receiveUpdate(ItemStateEvent updateEvent) {
 
     private <T extends Type> @Nullable T toAcceptedType(T originalType, Channel channel,
             Function<@Nullable String, @Nullable List<Class<? extends T>>> acceptedTypesFunction) {
+        String acceptedItemType = channel.getAcceptedItemType();
```

Comment:
The acctepted item type has been stored in the local variable `acceptedItemType` here that is used in the check in the following line. Why not using that local variable in `acceptedTypesFunction.apply(channel.getAcceptedItemType())` too instead of calling `getAcceptedItemType()` again?

Reply:
Makes sense and this change has already been done on #5734.